lib/Fmt_ast.ml

```diff
@@ -4412,15 +4412,112 @@ let fmt_repl_file c _ itms =
 
 (** Entry points *)
 
+let disabling (c : c) attr =
+  (not c.conf.opr_opts.disable)
+  && (update_config ~quiet:true c [attr]).conf.opr_opts.disable
+
+let enabling (c : c) attr =
+  c.conf.opr_opts.disable
+  && not (update_config ~quiet:true c [attr]).conf.opr_opts.disable
+
+let split_format_chunks (type a) c ctx (l : a list) ~is_disabling
+    ~is_enabling ~fmt_item ~loc_end =
+  let update c disable =
+    let opr_opts = {c.conf.opr_opts with disable} in
+    {c with conf= {c.conf with opr_opts}}
+  in
+  let groups =
+    let ref_c = ref c in
+    List.group l ~break:(fun x y ->
+        match (x, y) with
+        | _, k when Option.is_some (is_disabling !ref_c k) -> true
+        | _, k when Option.is_some (is_enabling !ref_c k) -> true
+        | k, _ when Option.is_some (is_disabling !ref_c k) ->
+            ref_c := update !ref_c true ;
+            false
+        | k, _ when Option.is_some (is_enabling !ref_c k) ->
+            ref_c := update !ref_c false ;
+            false
+        | _ -> false )
+  in
+  List.foldi groups ~init:(c, noop) ~f:(fun i (c, output) g ->
+      match g with
+      | k :: _ when Option.is_some (is_disabling c k) ->
```

Comment:
The disabled/enabled information could be stored in the group.

Reply:
done